Neos.Flow/Classes/Http/Headers.php

```diff
@@ -186,8 +202,9 @@ public function getAll()
     {
         $fields = $this->fields;
         $cacheControlHeader = $this->getCacheControlHeader();
-        if (!empty($cacheControlHeader)) {
-            $fields['Cache-Control'] = [$cacheControlHeader];
+        $fields['Cache-Control'] = [$cacheControlHeader];
+        if (empty($cacheControlHeader)) {
```

Comment:
I guess this needs a longer explanation.... Eventually we should return ALL headers, right now our Headers implementation handles two headers differently. One is `Set-Cookie` the other is `Cache-Control`. As you can see above I initialise `Cache-Control` empty in the defaults, therefore at this point it could've been empty all along or from setting it, either way, in that case it must be unset. Therefore unconditionally setting and then unsetting if empty, everything else would be more code.

I didn't include Set-Cookie here yet, for sake of backwards compat (seemed riskier than cache control).

Reply:
Ah, indeed. Could be this, I think, but fine either way:

        $fields = $this->fields;
        $fields['Cache-Control'] = [$this->getCacheControlHeader()];
        if (empty($fields['Cache-Control'])) {
            unset($fields['Cache-Control']);
        }
        return $fields;

And cookies are simply not returned by `getAll()` now, right? And that is like it is now?